core/chains/evm/txmgr/txmgr.go

```diff
@@ -81,7 +79,6 @@ type Txm[
 	txStore          txmgrtypes.TxStore[ADDR, CHAIN_ID, TX_HASH, BLOCK_HASH, R, SEQ, FEE, ADD]
 	db               *sqlx.DB
 	q                pg.Q
-	ethClient        evmclient.Client
```

Comment:
So now that we don't need ethClient in Txm, you could remove that input param ethClient from NewTxm(), to make the dependency more clear.
I think that fn only needs chainId, which we can directly pass in.
It will indicate that Txm has no dependency on any ethClient

Reply:
yup! that will be addressed in a follow-up PR for generalizing the top-level txmgr :)